hackrf: v2014.08.1

fdb opened the pull request:
I propose adding the HackRF library to Homebrew. These are the drivers and libraries for the HackRF device, a low-cast general-purpose software-defined radio.

An issue is that the test code executes `hackrf_info`, which fails if you don't have a HackRF device connected. All of the other installed programs have the same issue. Should I use a different test for this?

mikemcquaid commented:
> All of the other installed programs have the same issue. Should I use a different test for this?

Hmm, yeh. If there's a `--version` or `--help` you can use instead that's preferable to a test that'll fail. Thanks!

mikemcquaid commented on Library/Formula/hackrf.rb:
```diff
@@ -0,0 +1,22 @@
+require "formula"
```
Don't need this line any more, thanks.

fdb commented:
I've checked but all built-in bins return a non-zero exit code without a device. I've switched it to `pkg-config libhackrf --cflags`, which is probably what you want to do.

However, that means you need `pkgconfig` as well, which I added as a dependency.

mikemcquaid commented:
Non-zero output is OK. We can use `shell_output` which takes an argument to assert on a different exit code.

fdb commented:
OK, do you prefer that over the `pkgconfig` one?

mikemcquaid commented:
Yeh, please. Sorry!

fdb commented:
No problem at all! Here you go.

mikemcquaid commented:
Seems to fail to install, did it work for you locally? http://bot.brew.sh/job/Homebrew%20Pull%20Requests/22664/version=yosemite/testReport/junit/brew-test-bot/yosemite/install_hackrf/

mikemcquaid commented on Library/Formula/hackrf.rb:
```diff
@@ -0,0 +1,20 @@
+class Hackrf < Formula
+  homepage "https://github.com/mossmann/hackrf"
+  url "https://github.com/mossmann/hackrf/archive/v2014.08.1.tar.gz"
+  version "2014.08.1"
```
You can remove this, it's detected from the URL.

fdb commented:
Hey @mikemcquaid it looks like HackRF does require `pkg-config`, otherwise it can't find `libusb`. I've added it as a build dependency, it should build now.

mikemcquaid commented:
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format for new formulae is `hackrf 2014.08.1 (new formula)`.

fdb commented:
You're welcome :-)
